api.md

```diff
@@ -778,17 +778,37 @@ Return the same Variant object for chaining.
 ```
 
 
-#### respondWithFile(filePath)
-* ***filePath***: the path to the file to serve out (any route variables can be used in the file path as well)
+#### respondWithFile(options)
+* ***options***: options to override the default values. *(These options can also be globally provided in the smocks core options when starting the server, see ([Starting the server](#section/Concepts/Starting%2520the%2520server))*
+  * ***mockDir***: The base directory where the mocking data exists. 
+    * *default value set to* ```./mocked-data```
+  * ***fileName***: The file name smocks will search for in the ***mockDir*** directory structure and then set as the response.
+    * *default value set to* ```response.json```
+  * ***statusCode***: The status code of the response.
+    * *Default value set to* ```200```
+    
+
+This method enables you to have your json files in a route similar directory structure as follows. 
 
 Remember that using ```./``` will refer to the top level module directory (the directory where ```node_modules``` exists regardless of the location of the file that is referring to a file location with ```./```);
 
+Example 1
 ```javascript
     var smocks = require('smocks');
-    smocks.route({path: '/customer/{id}'}).respondWithFile('./customer-{id}.json')
+    smocks.route({path: '/customer/{id}'}).respondWithFile()
     .start(...)
 ```
 
-This would cause a request to ```/customer/1``` to return the file ```./customer-1.json```
+This would cause a request to ```/customer/123``` to return the file ```./mocked-data/customer/default/response.json``` where ```default```is the variant id, the dynamic ```{id}``` in the path will not be used here.
 
-Return the same Variant object for chaining.
+Example 2
+
+```javascript
+    var smocks = require('smocks');
+    smocks.route({path: '/customer/{id}'}).variant({id: 'invalid_customerid'}).respondWithFile({mockDir: './mocks, fileName: 'variantfile.json', statusCode: 422})
+    .start(...)
+```
+
+This would cause a request to ```/customer/123``` on the variant *invalid_customerid* to return the file ```./mocked-data/customer/invalid_customerid/response.json``` where ```default```is the variant id, the dynamic ```{id}``` in the path will not be used here.
+
+If the ```response.json``` file doesn't exist in the variant directory, smocks will search for it in the parent directory (i.e. ```./mocked-data/customer/response.json```) and use it for both *example 1* and *example 2* above. This is useful if you want to respond with the same file for all your variants on the same path. 
```

Comment:
I'm not sure that we really need a variant directory with the file lookup paths from my earlier comment.